src/App.tsx

```diff
@@ -15,49 +16,122 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+function getReorderedGoods(
+  goods: string[],
+  sortType: SortType,
+  isReversed: boolean,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((g1, g2) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return g1.localeCompare(g2);
+      case SortType.LENGTH:
+        return g1.length - g2.length;
+      default:
+        return 0;
+    }
+  });
+
+  return isReversed
+    ? visibleGoods.reverse()
+    : visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const alphabet = () => setSortType(() => SortType.ALPHABET);
+
+  const length = () => setSortType(() => SortType.LENGTH);
+
+  const reverse = () => setReversed(() => !isReversed);
```

Comment:
but here you do need the previous to set the next value
the only thing is that the functional set state accepts the prev state as an argument, you need to use it, otherwise right now you are still not using the prev state value properly

```suggestion
  const reverse = () => setReversed((isReversed) => !isReversed);
```